dashboards/resource_lakeview_dashboard.go

```diff
@@ -0,0 +1,179 @@
+package dashboards
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"log"
+
+	"github.com/databricks/databricks-sdk-go/service/dashboards"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Calculates MD5 hash of the given content
+func calculateMd5Hash(content []byte) string {
+	return fmt.Sprintf("%x", md5.Sum(content))
+}
+
+// Reads content from a JSON string or a file path and returns the content and its MD5 hash
+func readSerializedJsonContent(jsonStr, filePath string) (serJSON string, md5Hash string, err error) {
+	var content []byte
+	if filePath != "" {
+		content, err = common.ReadFileContent(filePath)
+		if err != nil {
+			return "", "", err
+		}
+	} else {
+		log.Printf("[INFO] Reading `serialized_json` of %d bytes", len(jsonStr))
+		content = []byte(jsonStr)
+	}
+	md5Hash = calculateMd5Hash(content)
+	return string(content), md5Hash, nil
+}
+
+// ResourceLakeviewDashboard manages lakeview dashboards
+func ResourceLakeviewDashboard() common.Resource {
+	s := common.StructToSchema(dashboards.Dashboard{},
```

Comment:
I think it'd be better to create a type and implement the methods like this one [here](https://github.com/databricks/terraform-provider-databricks/blob/6371eeab97f5b6cea348ee8e30c74c85b7d50c9d/jobs/resource_job.go#L485-L511). This is our latest convention for doing things.

Reply:
Schema custom function cannot be applied if the Schema struct is of non-local type

Reply:
<img width="841" alt="Screenshot 2024-07-08 at 07 27 45" src="https://github.com/databricks/terraform-provider-databricks/assets/171924202/56b5725b-815c-4642-a075-4cf13b689083">